Libraries/Lists/SectionList.js

```diff
@@ -322,14 +324,40 @@ class SectionList<SectionT: SectionBase<any>> extends React.PureComponent<
     }
   }
 
+  constructor(props: Props<SectionT>) {
+    super(props);
+    this._checkProps(this.props);
+  }
+
+  componentDidUpdate(prevProps: Props<SectionT>) {
+    this._checkProps(this.props);
+  }
+
+  _checkProps(props: Props<SectionT>) {
+    const {getItem, getItemCount, getItemParam} = props;
```

Comment:
Can you add a check to only do this with `__DEV__` and use `warning` instead of `invariant`?